maven-plugin/src/main/java/org/openmrs/maven/plugins/Clone.java

```diff
@@ -0,0 +1,206 @@
+
+    private String extractGitHubHttpKeyFromModulePom(String artifactId, String version, String groupId) throws MojoExecutionException, IOException {
+        downloadModulePom(new Artifact(artifactId, version, groupId, "pom"));
+
+        File pomDir = new File("pom/");
+        File pom = new File(pomDir ,this.artifactId+"-"+version+".pom");
+
+        BufferedReader br = null;
+        br = new BufferedReader(new FileReader(pom));
+        String url = "";
+        String line = br.readLine();
+        while (line != null) {
+            if (line.contains("https://github.com/")) {
+                url = line.substring(line.indexOf(">")+1, line.indexOf("</"));
+                break;
+            }
+                line = br.readLine();
+        }
+        //Cleanup
+        br.close();
+        pom.delete();
+        pomDir.delete();
+
+        return StringUtils.removeEnd(url, "/") + ".git";
+    }
+
+    private void downloadModulePom(Artifact artifact) throws MojoExecutionException {
+        MojoExecutor.Element[] artifactItems = new MojoExecutor.Element[1];
+        artifactItems[0] = artifact.toElement("pom/");
+        List<Element> configuration = new ArrayList<Element>();
+        configuration.add(element("artifactItems", artifactItems));
+        executeMojo(
+                plugin(
+                        groupId(SDKConstants.PLUGIN_DEPENDENCIES_GROUP_ID),
+                        artifactId(SDKConstants.PLUGIN_DEPENDENCIES_ARTIFACT_ID),
+                        version(SDKConstants.PLUGIN_DEPENDENCIES_VERSION)
+                ),
+                goal(GOAL_COPY),
+                configuration(configuration.toArray(new Element[0])),
+                executionEnvironment(mavenProject, mavenSession, pluginManager)
+        );
+    }
+
+    private void forkRepo(String repoToForkName) {
+
+        System.out.println("Forking repository:");
+        String repoToForkOwner = "openmrs";
+        GitHubClient client = new GitHubClient();
+        System.out.print("- Client authentication");
+        client.setCredentials(githubUsername, githubPassword);
+        System.out.println(" - DONE");
+        RepositoryService service = new RepositoryService();
+
+        System.out.print("- Service authentication");
+        service.getClient().setCredentials(githubUsername, githubPassword);
+        System.out.println(" - DONE");
+        RepositoryId toBeForked = new RepositoryId(repoToForkOwner, repoToForkName);
+        try {
+            System.out.print("- Forking");
+            service.forkRepository(toBeForked);
+            System.out.println(" - DONE");
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new IllegalStateException("Failed to fork repository \"" + repoToForkOwner + "/" + repoToForkName+ "\"");
+        }
+    }
+
+    private void cloneRepo(String upstreamHttpKey) {
+
+        String originHttpKey = upstreamHttpKey.replace(OPENMRS_GITHUB_KEY_PARTIAL, "/" + githubUsername + "/");
```

Comment:
We can easily make it work without having to hardcode /openmrs/.

How about you use regular expression to replace github.com/any_owner/?